lib/matplotlib/figure.py

```diff
@@ -1291,6 +1354,65 @@ def subplots_adjust(self, left=None, bottom=None, right=None, top=None,
                 ax._set_position(ax.get_subplotspec().get_position(self))
         self.stale = True
 
+    def set_subplotpars(self, subplotparams={}):
```

Comment:
Default values should not be mutable objects like a dictionary. You should set it to `None`, and then check that in the function.
```suggestion
    def set_subplotpars(self, subplotparams=None):
```